### PR TITLE
Walk point

### PR DESCRIPTION
**Pull request by lxgr-linux**
Enhanced NPC abilities.

### CODE REVIEW

File: pokete_classes/npcs.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        if not self.check_walk(x, y):
+            return False
+        for i in vec.obs:
+            self.set(i.rx + o_x, i.ry + o_y)
+            self.mvmp.full_show()
+            time.sleep(0.2)
+        return True
+
+    def give(self, name, item):
+        """Method thats gifts an item to the player"""
```
Comment (MaFeLP) [inline pokete_classes/npcs.py:106]:
```suggestion
        """Method that gifts an item to the player"""
```

File: pokete_classes/npcs.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        self.will = False
+        self.used_npcs.append(self.name)
+        if ask_bool(self._ev, self.mvmp,
+                    f"{name} gifted you a '{item.pretty_name}'. \
+Do you want to accept it?"):
+            self.fig.give_item(item.name)
+
+
+class Trainer(NPC):
+    """Trauner class to fight against"""
```
Comment (MaFeLP) [inline pokete_classes/npcs.py:117]:
```suggestion
    """Trainer class to fight against"""
```

File: pokete_classes/npcs.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    def action(self):
+        """Interaction with the trainer"""
+        o_x = self.x
+        o_y = self.y
+        if self.fig.has_item("shut_the_fuck_up_stone"):
+            return
+        if self.fig.x == self.x and self.poke.hp > 0 \
+                and (self.name not in self.used_npcs
+                    or not self.settings.save_trainers) \
+                and self.check_walk(self.fig.x, self.fig.y):
```
Comment (MaFeLP) [inline pokete_classes/npcs.py:149]:
```suggestion
        if self.fig.x == self.x and self.poke.hp > 0 \
                and (self.name not in self.used_npcs
                     or not self.settings.save_trainers) \
                and self.check_walk(self.fig.x, self.fig.y):
```

File: pokete_classes/npcs.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            self.mvmp.full_show()
+            time.sleep(0.7)
+            self.exclamate()
+            self.walk_point(self.fig.x, self.fig.y)
+            if any(poke.hp > 0 for poke in self.fig.pokes[:6]):
+                self.text(self.texts)
+                winner = self.fight([poke for poke in self.fig.pokes[:6]
+                                    if poke.hp > 0][0],
+                               self.poke, info={"type": "duel", "player": self})
```
Comment (MaFeLP) [inline pokete_classes/npcs.py:158]:
```suggestion
                winner = self.fight([poke for poke in self.fig.pokes[:6]
                                    if poke.hp > 0][0],
                                    self.poke,
                                    info={"type": "duel", "player": self})
```

File: pokete_classes/npcs.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                self.text({True: self.lose_texts,
+                           False: self.win_texts + [" < Here u go 20$"]}
+                                [winner == self.poke])
```
Comment (MaFeLP) [inline pokete_classes/npcs.py:161]:
```suggestion
                          [winner == self.poke])
```

**Review by MaFeLP — CHANGES_REQUESTED** [review]

File: pokete_classes/npcs.py (hunk 5 above)
Comment (lxgr-linux) [reply pokete_classes/npcs.py:163]: This is indented, to show that it's not another argument for `self.text`, but rather the indexation of the dict before.

**Review by MaFeLP — APPROVED** [review]

**Pull request merged**